File: src/main/java/postBox/abstractBox.java
package postBox;

public  abstract class abstractBox implements Box{

    float length;
    float width ;
    int   tall  ;

    public abstractBox(Float length, Float width, int tall) {
        this.length = length;
        this.width = width;
        this.tall = tall;
    }

    abstract public boolean validate(float length, float width, int tall);

    public boolean check(float length, float width, int tall){
        if(this.length >= length && this.width >= width && this.tall >= tall)
            return true;
        else
            return false;
    }
}

File: src/main/kotlin/postBoxK/abstractBox.kt
package postBoxK



abstract class abstractBox(var length: Float, var width: Float, var tall: Int) : Box {

    abstract override fun validate(length: Float, width: Float, tall: Int): Boolean

    override fun check(length: Float, width: Float, tall: Int): Boolean {
       return if(  this.length >= length && this.width >= width && this.tall >= tall ) true else false
    }

}
